api/api/admin/__init__.py

```diff
@@ -154,23 +154,40 @@ class IndividualUserPreferencesAdmin(admin.ModelAdmin):
     currently logged-in user's preferences
     """
 
-    can_delete = False
     verbose_name_plural = "My Preferences"
     verbose_name = "My Preferences"
     form = UserPreferencesAdminForm
 
+    def has_add_permission(self, request):
+        """Remove functionality to add new ``UserPreferences`` objects."""
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        """Remove functionality to delete ``UserPreferences`` objects."""
+        return False
+
     def get_queryset(self, request):
+        """
+        Restrict the user's ability to view and change only the
+        ``UserPreferences`` instance mapped to their own ``User`` instance.
+
+        :return: the queryset of just the logged-in user's preferences
+        """
+
         qs = super().get_queryset(request)
         return qs.filter(user=request.user)
 
-    def has_change_permission(self, request, obj=None):
-        if not obj:
-            # the changelist itself
-            return True
-        return obj.user == request.user
-
     def changelist_view(self, request, extra_context=None):
-        obj = self.get_queryset(request).first()
+        """
+        Redirect to the change form for the logged-in user's preferences. This
+        view automatically creates the ``UserPreferences`` instance, if one does
+        not already exist for the current user.
+        """
+
+        if (qs := self.get_queryset(request)).exists():
+            obj = qs.first()
+        else:
+            obj = UserPreferences.objects.create(user=request.user)
```

Comment:
Two things here. We have a signal to create these, and ideally we would only have this or the signal, not both. One way and only one way to understand when and how this model is created.

The signal is much better, though, because it ensures the model exists when User is created, which is necessary because this approach only creates it if the user actually tries to update their preferences: meaning, when we try to retrieve the preferences and use them, we don't have to worry about whether they exist. The signal ensures they exist. This approach does not, and is redundant with the signal. I don't think this is a good idea and it will be confusing when trying to understand where and when the user preferences instance creation is intended to happen.

Reply:
That's fair, I had added this as a fallback to avoid the case I experienced locally where I deleted the `UserPreferences` for the current user from the Django shell and was stuck.

- Couldn't create a new `UserPreference` instance from the UI because of removed 'add' permission.
- Couldn't navigate to the form because `.first()` returned `None` and then we tried to access `.id` on it.

This was a local case though, so I concede. Removed in 1f0d8b0.